docs/data/toolpad/building-ui/custom-components.md

```diff
@@ -0,0 +1,31 @@
+# Custom components
+
+<p class="description">When built in components are not enough to achieve your wanted results build your own using custom components UI</p>
+
+## Creating new component
+
+In order to create a new custom component locate **+** button in the Explorer near Components section and click it:
+
+<img src="/static/toolpad/docs/custom-component-1.png" width="241px" alt="Custom component" />
+
+Choose the name for your component and click **Create**:
+
+<img src="/static/toolpad/docs/custom-component-2.png" width="460px" alt="Custom create" />
+
+You can now use code editor (on the left) and live preview (on the right) to define your custom component:
+
+<img src="/static/toolpad/docs/custom-component-3.png" width="990px" alt="Custom component IDE" />
+<br />
+<br />
+
+Supported features:
+
+- Expose props API through `argTypes` definition when using `createComponent` method.
```

Comment:
Maybe not needed for now but when it's more set in stone we can document what sort of things can be done with `argTypes`.

Reply:
I'd leave it for now, otherwise there is no way for users to know how to expose props 🤔